src/xgov-dapp/src/shared/ProposalCard.tsx

```diff
@@ -28,14 +28,22 @@ export const ProposalCard = ({
   hasClosed = false,
 }: ProposalCardProps) => {
   // Handle collapse state
-  const descriptionRef = useRef(null)
-  const isOverflow = useOverflow(descriptionRef)
+  const [isOverflow, setIsOverflow] = useState(false)
+  const { ref } = useOverflow((result) => {
+    if (result !== isOverflow) setIsOverflow(result)
+  })
+  const [hasOpened, setHasOpened] = useState(false)
```

Comment:
Why do we need `hasOpened`?

Reply:
When it is open, it is no longer overflowing the container. The `Show More/Show Less` button is only displayed when it's overflowing, if it's ever been opened we can safely display the button